Update: deprecate personal config (fixes #11914, refs eslint/rfcs#32) (#12426)
eslint/eslint: docs/user-guide/configuring.md

```

This message occurs because ESLint is unsure if you wanted to actually lint the file or not. As the message indicates, you can use `--no-ignore` to omit using the ignore rules.

## Personal Configuration File (deprecated)

> ⚠️ **This feature has been deprecated**. It will start to print deprecation warnings on ESLint `7.0.0` and will be removed on ESLint `8.0.0`. If you want to continue to use personal configuration file, please use [`--config` CLI option](https://eslint.org/docs/user-guide/command-line-interface#-c---config). If you want to know the discussion of this decision, see [RFC 28](https://github.com/eslint/rfcs/pull/28) and [RFC 32](https://github.com/eslint/rfcs/pull/32).

Reviewer: ```suggestion
⚠️ **This feature has been deprecated**. ESLint will start to print deprecation warnings starting with the `7.0.0` release. These warnings will then be removed in the `8.0.0` release. If you want to continue to use personal configuration files, please use the [`--config` CLI option](https://eslint.org/docs/user-guide/command-line-interface#-c---config). For more information regarding this decision, please see [RFC 28](https://github.com/eslint/rfcs/pull/28) and [RFC 32](https://github.com/eslint/rfcs/pull/32).
```

Author: Thank you for your suggestion. I have some questions.

> ESLint will start to print deprecation warnings starting with the `7.0.0` release.

Are not "start to" and "starting with" redundant?

> These warnings will then be removed in the `8.0.0` release. 

What we will remove in `8.0.0` is the personal configuration file functionality. (sure, the warnings will be removed as well, but...). Does it make sense if we clarify what's removed?

Reviewer: > Are not "start to" and "starting with" redundant?

Agreed that doesn't flow very well. 

> What we will remove in 8.0.0 is the personal configuration file functionality. (sure, the warnings will be removed as well, but...). Does it make sense if we clarify what's removed?

Ah yeah, I misunderstood that. Given the current sentence structure, it looks like it's implicitly referring to the printing behavior being removed.

Maybe `ESLint will print deprecation warnings beginning with the 7.0.0 release. The feature and deprecation warnings will then be removed in the 8.0.0 release.`?

> Here `~/` means [the home directory of the current user on your OS](https://nodejs.org/api/os.html#os_os_homedir). The personal configuration file is `~/.eslintrc.*` file. ESLint handles `~/.eslintrc.*` file in special.

Reviewer: ```suggestion
`~/` refers to [the home directory of the current user on your preferred operating system](https://nodejs.org/api/os.html#os_os_homedir). The personal configuration file being referred to here is `~/.eslintrc.*` file, which is currently handled differently than other configuration files.
```

### How ESLint Finds Personal Configuration File

If `eslint` could not find any configuration file in the project, `eslint` loads `~/.eslintrc.*` file.

If `eslint` could find configuration files in the project, `eslint` ignores `~/.eslintrc.*` file even if it's in an ancestor directory of the project directory.

### How Personal Configuration File Behaves

> `~/.eslintrc.*` file behaves samely as regular configuration files, but there are some notes.

Reviewer 2: ```suggestion
`~/.eslintrc.*` file behaves in the same way as regular configuration files, but there are some notes.
```

> `~/.eslintrc.*` file behaves in the same way as regular configuration files, but there are some notes.

Reviewer: ```suggestion
`~/.eslintrc.*` files behave similarly to regular configuration files, with some exceptions:
```

> The `~/.eslintrc.*` file loads shareable configs and custom parsers from `~/node_modules/` as similar to `require()` in home directory. Please mind that it doesn't load global-installed packages.

Reviewer: ```suggestion
`~/.eslintrc.*` files load shareable configs and custom parsers from `~/node_modules/` – similarly to `require()` – in the user's home directory. Please note that it doesn't load global-installed packages.
```

> The `~/.eslintrc.*` file loads plugins from `$CWD/node_modules` because ESLint loads plugins from `$CWD/node_modules` at all in order to identify plugins uniquely. If you want to use plugins with `~/.eslintrc.*` file, you have to install the plugins into each project. Or you can use [`--resolve-plugins-relative-to` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--resolve-plugins-relative-to) to change the location where ESLint loads plugins from.

Reviewer 2: ```suggestion
The `~/.eslintrc.*` file loads plugins from `$CWD/node_modules` because ESLint loads plugins from `$CWD/node_modules` by default in order to identify plugins uniquely. If you want to use plugins with `~/.eslintrc.*` file, you have to install the plugins into each project. Or you can use [`--resolve-plugins-relative-to` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--resolve-plugins-relative-to) to change the location where ESLint loads plugins from.
```

> The `~/.eslintrc.*` file loads plugins from `$CWD/node_modules` because ESLint loads plugins from `$CWD/node_modules` by default in order to identify plugins uniquely. If you want to use plugins with `~/.eslintrc.*` file, you have to install the plugins into each project. Or you can use [`--resolve-plugins-relative-to` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--resolve-plugins-relative-to) to change the location where ESLint loads plugins from.

Reviewer: ```suggestion
`~/.eslintrc.*` files load plugins from `$CWD/node_modules` by default in order to identify plugins uniquely. If you want to use plugins with `~/.eslintrc.*` files, plugins must be installed locally per project. Alternatively, you can use the [`--resolve-plugins-relative-to` CLI option](https://eslint.org/docs/user-guide/command-line-interface#--resolve-plugins-relative-to) to change the location from which ESLint loads plugins.
```
